pkg/nsm/apis/internsm/internsm.proto

```diff
@@ -0,0 +1,105 @@
+syntax = "proto3";
+
+package internsm;
+
+import "github.com/ligato/networkservicemesh/pkg/nsm/apis/common/common.proto";
+
+// InterNsm specifies how Network Service Managers (NSMs) communicate with each other
+// to manage a RemoteConnection between a Network Service Clients (NSC) and Network Service Endpoints (NSE)
+service InterNsm {
+    // RequestRemoteConnection - Service called by one NSM to request a RemoteConnection to a Network Service Endpoint from another NSM
+    rpc RequestRemoteConnection (RemoteConnectionRequest) returns (RemoteConnectionReply);
+    // DeleteRemoteConnection - Service called by one NSM to to request a RemoteConnection be deleted
+    rpc DeleteRemoteConnection (RemoteConnection) returns (RemoteConnectionReply);
+    // UpdateRemoteConnection - Service called by one NSM to request update of a RemoteConnection by another NSM.
+    // Note: Only intended to Update ConnectionContext
+    rpc UpdateRemoteConnection (RemoteConnection) returns (RemoteConnectionReply);
+    // MonitorRemoteConnections - Service called by one NSM to request streaming updates on existing RemoteConnections
+    //     RemoteConnectionId - interpreted as a 'selector' with empty fields being wildcarded.
+    //                          Example: specifying source_nsm_id will cause streaming of updates for all connections matching source nsm_id
+    rpc MonitorRemoteConnections (RemoteConnectionId) returns (stream RemoteConnectionUpdate);
+}
+
+// RemoteConnectionId - uniquely identifies the connection
+// Note: A connection in Requested state may have a partially specified RemoteConnectionId.  A RemoteConnectionId must be fully specified in CONNECTED state
+message RemoteConnectionId {
+    // source_nsm_id - Id of the source NSM - provided authoritatively by source NSM in RemoteConnectionRequest
+    // Can be used for 'recovery' in the event of an NSM restart by 'Monitoring' all RemoteConnections
+    // from its peers matching its own source_nsm_id
+    string source_nsm_id = 1;
```

Comment:
What is the id? I mean hostname, pod name or something else?

Reply:
@sbezverk Its up the nsm to generate.  It would be good if it were something durable in the event of NSM restart, so perhaps node_name etc.

Please do note: not all NSMs will be the K8s node NSMs, so the source_nsm_id can be something like  node name, pod id, etc for the K8s node NSMs... but it can't be *required* to be that in the InterNSM API.

Reply:
@edwarnicke Is this to handle the eNSM case as well as the case where NSM works with orchestration systems other than Kubernetes?

Reply:
@mestery Short answer: Yes.  K8s-isms need to be kept out of the InterNSM API, otherwise we cannot have eNSMs or cross cluster etc.  That doesn't mean that a K8s related thing can't be used in that field, just that the field itself isn't defined as a K8s thing.

Reply:
@edwarnicke I was just trying to understand where these ID comes from, and how to enforce consistency.